### PR TITLE
#15 add field tags

### PR DESCRIPTION
**Pull request by as27**
Added support for struct tag tableheader.
Tests and documentation added with examples.

### CODE REVIEW

File: schema/schema_test.go [hunk 1]
```diff
@@ -40,6 +40,33 @@ func ExampleSchema_Decode() {
 	// {Name:Bar Age:43}
 }
 
+func ExampleSchema_Decode_withTags() {
```
Comment (danielfireman) [inline schema/schema_test.go]: I would suggest for that being the actual ExampleSchema_Decode (removing the method above) and add a small comment at line 55 saying that removing the struct tag will make the field name match the column name. What do you think?

Same for encoding, decode/encoding table.

What do you think?

Also, could you please add unit tests for struct tags to TestDecode/TestEncode bellow? We never know when those examples are going to change :D

File: schema/schema.go [hunk 2]
```diff
@@ -196,7 +196,11 @@ func (s *Schema) Decode(row []string, out interface{}) error {
 		fieldValue := outv.Field(i)
 		if fieldValue.CanSet() { // Only consider exported fields.
 			field := outt.Field(i)
-			f, fieldIndex := s.GetField(field.Name)
+			fieldName, ok := field.Tag.Lookup("tableheader")
```
Comment (danielfireman) [inline schema/schema.go]: Could you please make tableheader a constant?

File: schema/schema_test.go (hunk 1 above)
Comment (as27) [reply schema/schema_test.go]: Yes. Good point.

File: schema/schema.go (hunk 2 above)
Comment (as27) [reply schema/schema.go]: Also thought about this.

File: schema/schema_test.go [hunk 3]
```diff
@@ -22,11 +22,13 @@ func ExampleSchema_Decode() {
...
-	// And we would like to process them using Go types. First we need to create a struct to hold the
-	// content of each row.
+	// And we would like to process them using Go types. First we need to create a struct to
+	// hold the content of each row.
+	// The tag tableheader maps the field to the schema. If no tag is set the name of the field
+	// has to be the same like inside the schema.
 	type person struct {
-		Name string
-		Age  int
+		MyName string `tableheader:"Name"`
```
Comment (danielfireman) [inline schema/schema_test.go]: Nice!

File: schema/schema_test.go [hunk 4]
```diff
@@ -255,6 +262,22 @@ func TestSchema_Decode(t *testing.T) {
 			t.Errorf("value:Age want:42 got:%d", t1.Age)
 		}
 	})
+	t.Run("StructWithTags", func(t *testing.T) {
```
Comment (danielfireman) [inline schema/schema_test.go]: Could you please add an analogous test to TestSchema_Encode?

Comment (as27) [reply schema/schema_test.go]: Oh yes, just missed this. Now the tests are complete

**Pull request merged**